adobe/aem-core-cif-components
# CIF-1787 - Implement ComponentExporter for CIF teaser component (the …

## Description

* Add the required annotations to the CommerceTeaserImpl Sling Model
* Update the action items to include the `categoryId` / `productSlug` to be used by the SPA

## Related Issue

CIF-1787

## Motivation and Context


Export the Sling Model data as JSON 

## How Has This Been Tested?

Unit tests and functional testing





Steps for functional testing:

1/ Install AEM Venia and this branch of the CIF Core Components project
2/ Open the home page in the editor and add a Teaser component, 
3/ Configure the Teaser component as follows:
* Add an asset
* Add a title and a description
* Add a call to action pointing to a category
* Add a CTA pointing to a product
4/ Open the model JSON using the model selector, i.e. `http://localhost:4502/content/venia/us/en.model.json`
5/ Look for the teaser component and check the following:
* The `actions` JSON snippet should be an array
* An `action` should at least have the title and the `identifier` object. For example, for a category, an action could look like this
```JSON
{
  "title": "Shop now",
  "url": "/content/venia/us/en/products/category-page.3.html",
  "entityIdentifier": {
    "entityType": "CATEGORY",
    "value": "3",
    "type": "ID"
  }
}
``` 
* Check that the following properties are exported: title, description, pretitle, asset path.
* Check that the resource type is `core/cif/components/content/teaser/v1/teaser`
